src/Riok.Mapperly/Descriptors/MappingBodyBuilders/NewValueTupleMappingBodyBuilder.cs

```diff
@@ -0,0 +1,201 @@
+
+                return false;
+            }
+
+            // nullability is handled inside the member expressionMapping
+            var paramType = targetMember.Type.WithNullableAnnotation(targetMember.NullableAnnotation);
+            var delegateMapping =
+                ctx.BuilderContext.FindMapping(sourcePath.MemberType, paramType)
+                ?? ctx.BuilderContext.FindOrBuildMapping(sourcePath.Member.Type.NonNullable(), paramType.NonNullable());
+
+            if (delegateMapping == null)
+            {
+                ctx.BuilderContext.ReportDiagnostic(
+                    DiagnosticDescriptors.CouldNotMapMember,
+                    ctx.Mapping.SourceType,
+                    sourcePath.FullName,
+                    sourcePath.Member.Type,
+                    ctx.Mapping.TargetType,
+                    targetMember.Name,
+                    targetMember.Type
+                );
+                return false;
+            }
+
+            if (delegateMapping.Equals(ctx.Mapping))
+            {
+                ctx.BuilderContext.ReportDiagnostic(
+                    DiagnosticDescriptors.ReferenceLoopInCtorMapping,
+                    ctx.Mapping.SourceType,
+                    sourcePath.FullName,
+                    ctx.Mapping.TargetType,
+                    targetMember.Name
+                );
+                return false;
+            }
+
+            var memberMapping = new NullMemberMapping(
+                delegateMapping,
+                sourcePath,
+                paramType,
+                ctx.BuilderContext.GetNullFallbackValue(paramType),
+                !ctx.BuilderContext.IsExpression
+            );
+
+            var ctorMapping = new ValueTupleConstructorParameterMapping(targetMember, memberMapping);
+            constructorParameterMappings.Add(ctorMapping);
+            mappedTargetMemberNames.Add(targetMember.Name);
+        }
+
+        return true;
+    }
+
+    private static bool TryFindConstructorParameterSourcePath(
+        INewValueTupleBuilderContext<INewValueTupleMapping> ctx,
+        IFieldSymbol field,
+        [NotNullWhen(true)] out MemberPath? sourcePath
+    )
+    {
+        sourcePath = null;
+
+        if (!ctx.MemberConfigsByRootTargetName.TryGetValue(field.Name, out var memberConfigs))
+        {
+            if (
+                MemberPath.TryFind(
+                    ctx.Mapping.SourceType,
+                    MemberPathCandidateBuilder.BuildMemberPathCandidates(field.Name),
+                    ctx.IgnoredSourceMemberNames,
+                    StringComparer.OrdinalIgnoreCase,
+                    out sourcePath
+                )
+            )
+                return true;
+
+            // if standard matching fails, use the underlying field
+            // if source is a tuple compare the underlying field ie, Item1, Item2
+            // if source is not a tuple compare top level members
+            if (ctx.Mapping.SourceType.IsTupleType)
```

Comment:
I'd invert this check and return early, probably even combine with the check on L140.